wasatch/SPIDevice.py

```diff
@@ -0,0 +1,331 @@
+        self.ready.direction = digitalio.Direction.INPUT
+
+        # Initialize D6 as the trigger
+        self.trigger = digitalio.DigitalInOut(board.D6)
+        self.trigger.direction = digitalio.Direction.OUTPUT
+        self.trigger.value = False
+
+        # Take control of the SPI Bus
+        while not self.SPI.try_lock():
+            pass
+
+        # Configure the SPI bus
+        self.SPI.configure(baudrate=20000000, phase=0, polarity=0, bits=8)
+        self.process_f = self._init_process_funcs()
+
+    def connect(self) -> SpectrometerResponse:
+        eeprom_pages = []
+        for i in range(EEPROM.MAX_PAGES):
+            response = bytearray(2)
+            while self.ready.value:
+                self.SPI.readinto(response,0,2)
+            page = self._EEPROMReadPage(i)
+            log.info(f"spi read page {i} with data {page}")
+            eeprom_pages.append(page)
+        eeprom_pages = [bytearray([val for val in page[9:74]]) for page in eeprom_pages]
+        self.settings.eeprom.parse(eeprom_pages)
+        self.settings.eeprom.active_pixels_horizontal = 1952
+        self.settings.state.integration_time_ms = 10
+        log.info("SPI connect done, returning True")
+        return SpectrometerResponse(True)
+
+    def disconnect(self) -> SpectrometerResponse:
+        self.disconnect = True
+        return SpectrometerResponse(True)
+
+    def acquire_data(self) -> SpectrometerResponse:
+        log.debug("spi starts reading")
+        if self.disconnect:
+            log.debug("disconnecting, returning False for the spectrum")
+            return False
+        averaging_enabled = (self.settings.state.scans_to_average > 1)
+        reading = Reading(self.device_id)
+
+        try:
+            reading.integration_time_ms = self.settings.state.integration_time_ms
+            reading.laser_power_perc    = self.settings.state.laser_power_perc
+            reading.laser_power_mW      = self.settings.state.laser_power_mW
+            reading.laser_enabled       = self.settings.state.laser_enabled
+            reading.spectrum            = self._Acquire()
+            if reading.spectrum == False:
+                return False
+        except usb.USBError:
+            self.failure_count += 1
+            log.error(f"SPI Device: encountered USB error in reading for device {self.device}")
+
+        if not reading.failure:
+            if averaging_enabled:
+                if self.sum_count == 0:
+                    self.summed_spectra = [float(i) for i in reading.spectrum]
+                else:
+                    log.debug("device.take_one_averaged_reading: summing spectra")
+                    for i in range(len(self.summed_spectra)):
+                        self.summed_spectra[i] += reading.spectrum[i]
+                self.sum_count += 1
+                log.debug("device.take_one_averaged_reading: summed_spectra : %s ...", self.summed_spectra[0:9])
+
+        if self.settings.eeprom.bin_2x2:
```

Comment:
I feel like bin_2x2 should happen *after* scan averaging.  Right now, it looks like you're applying 2x2 binning to the FIRST spectrum read, then you average a bunch of non-2x2-binned spectra into that.